core/src/main/java/org/mule/routing/filters/WildcardFilter.java

```diff
@@ -90,25 +89,34 @@ public boolean accept(Object object)
                     candidate = candidate.toLowerCase();
                 }
 
-                int i = pattern.indexOf('*');
-                if (i == -1)
+                int firstWildcardIndex = pattern.indexOf('*');
```

Comment:
Maybe it would be a nice refactor to separate the parsing of the pattern from the actual evaluation, for code clarity.